### PR TITLE
Allow setting `write.parquet.row-group-limit`

### LINKED ISSUE #1013: NotImplementedError: Parquet writer option(s) ['write.parquet.row-group-size-bytes'] not implemented
### Apache Iceberg version

0.7.0 (latest release)

### Please describe the bug 🐞

it was working fine, and today, I got this ? using Tabular as a catalog

```
/usr/local/lib/python3.10/dist-packages/pydantic/main.py:415: UserWarning: Pydantic serializer warnings:
  Expected `TableIdentifier` but got `dict` - serialized value may not be as expected
  return self.__pydantic_serializer__.to_json(
---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
<timed exec> in <module>

[/usr/local/lib/python3.10/dist-packages/pyiceberg/table/__init__.py](https://localhost:8080/#) in append(self, df, snapshot_properties)
   1572         """
   1573         with self.transaction() as tx:
-> 1574             tx.append(df=df, snapshot_properties=snapshot_properties)
   1575 
   1576     def overwrite(

3 frames
[/usr/local/lib/python3.10/dist-packages/pyiceberg/io/pyarrow.py](https://localhost:8080/#) in _get_parquet_writer_kwargs(table_properties)
   2288     ]:
   2289         if unsupported_keys := fnmatch.filter(table_properties, key_pattern):
-> 2290             raise NotImplementedError(f"Parquet writer option(s) {unsupported_keys} not implemented")
   2291 
   2292     compression_codec = table_properties.get(TableProperties.PARQUET_COMPRESSION, TableProperties.PARQUET_COMPRESSION_DEFAULT)

NotImplementedError: Parquet writer option(s) ['write.parquet.row-group-size-bytes'] not implemented
```

### PR DESCRIPTION
**Pull request by Fokko**
And update the docs

Fixes #1013

### CODE REVIEW

**Comment by sungwy** [comment]
LGTM @Fokko - merging in the change from main to resolve the conflict on the doc

**Review by sungwy — APPROVED** [review]

**Comment by Fokko** [comment]
Also threw in a test here 👍

**Review by sungwy — APPROVED** [review]

File: mkdocs/docs/configuration.md [hunk 1]
```diff
@@ -32,6 +32,7 @@ Iceberg tables support table properties to configure table behavior.
 | -------------------------------------- | --------------------------------- | ------- | ------------------------------------------------------------------------------------------- |
 | `write.parquet.compression-codec`      | `{uncompressed,zstd,gzip,snappy}` | zstd    | Sets the Parquet compression coddec.                                                        |
 | `write.parquet.compression-level`      | Integer                           | null    | Parquet compression level for the codec. If not set, it is up to PyIceberg                  |
+| `write.parquet.row-group-limit`        | Number of rows                    | 1048576 | The upper bound of the number of entries within a single row group                          |
```
Comment (zhongyujiang) [inline mkdocs/docs/configuration.md:35]: @Fokko @sungwy Thanks, I believe this has resolved my issue #1012 as well. 

However, I would like to remind you that this option already exists in the doc, right after `write.parquet.dict-size-bytes`, the UI doesn't allow me to leave a comment there, so please expand the collapsed area to see it.

Additionally, I'm kind of curious as to why the default value used this time is significantly larger than the previous one?

Comment (sungwy) [reply mkdocs/docs/configuration.md:35]: Thank you for flagging this @zhongyujiang - I'll get the second one below with the older default value removed.

To my understanding the new value is the correct default value that matches the default in the PyArrow ParquetWriter: https://arrow.apache.org/docs/python/generated/pyarrow.parquet.ParquetWriter.html

**Pull request merged**